src/linter/linter.ts

```diff
@@ -11,6 +11,11 @@ import type {AbstractReader, Resource} from "@ui5/fs";
 import ConfigManager, {UI5LintConfigType} from "../utils/ConfigManager.js";
 import {Minimatch} from "minimatch";
 
+// Internal analytical variable that will store matchers' count.
+// We cannot predict the outcome of the matchers, so stash usage statistics
+// and later analyze the results from it.
+const matchedPatterns = new Set<string>();
```

Comment:
Having this here as a singleton set will most probably cause issues when running multiple linting activities in the same process in parallel. We should think about moving it to an instance of the current linting activity, which might require some refactoring (e.g. turning this module into a class might help).